wasm/src/utilities/encrypt.rs

```diff
@@ -0,0 +1,257 @@
+
+        // Decrypt the owner.
+        let owner = match record_native.owner().is_public() {
+            true => record_native.owner().decrypt_with_randomizer(&[]).map_err(|e| e.to_string())?,
+            false => record_native.owner().decrypt_with_randomizer(&[randomizers[index]]).map_err(|e| e.to_string())?,
+        };
+
+        // Increment the index if the owner is private.
+        if record_native.owner().is_private() {
+            index += 1;
+        }
+
+        // Decrypt the program data.
+        let mut decrypted_data = IndexMap::with_capacity(record_native.data().len());
+        for (id, entry, num_randomizers) in
+            record_native.data().iter().map(|(id, entry)| (id, entry, Self::num_entry_randomizers(&entry)))
+        {
+            // Retrieve the result for `num_randomizers`.
+            let num_randomizers = num_randomizers? as usize;
+            // Retrieve the randomizers for this entry.
+            let randomizers = &randomizers[index..index + num_randomizers];
+            // Decrypt the entry.
+            let entry = match entry {
+                // Constant entries do not need to be decrypted.
+                CiphertextEntryNative::Constant(plaintext) => PlaintextEntryNative::Constant(plaintext.clone()),
+                // Public entries do not need to be decrypted.
+                CiphertextEntryNative::Public(plaintext) => PlaintextEntryNative::Public(plaintext.clone()),
+                // Private entries are decrypted with the given randomizers.
+                CiphertextEntryNative::Private(private) => PlaintextEntryNative::Private(
+                    PlaintextNative::from_fields(
+                        &private
+                            .iter()
+                            .zip_eq(randomizers)
+                            .map(|(ciphertext, randomizer)| *ciphertext - randomizer)
+                            .collect::<Vec<_>>(),
+                    )
+                    .map_err(|e| e.to_string())?,
+                ),
+            };
+            // Insert the decrypted entry.
+            if decrypted_data.insert(*id, entry).is_some() {
+                return Err(format!("Duplicate identifier in record: {}", id));
+            }
+            // Increment the index.
+            index += num_randomizers;
+        }
+
+        // Return the decrypted record.
+        let decrypted_record = RecordPlaintextNative::from_plaintext(owner, decrypted_data, *record_native.nonce())
+            .map_err(|e| e.to_string())?;
+
+        Ok(RecordPlaintext::from(decrypted_record))
+    }
+
+    #[wasm_bindgen(js_name = "generateTvk")]
+    pub fn generate_tvk(view_key: &ViewKey, tpk: &Group) -> Field {
+        tpk.scalar_multiply(&view_key.to_scalar()).to_x_coordinate()
+    }
+
+    /// Creates a record view key from the view key.  This method is intended to be used
+    /// by the record owner to enable decryption of a select record by a third party.
```

Comment:
```suggestion
    /// Creates a record view key from the view key.  This can be later be used to decrypt a record without revealing an account's view key.
```